packages/html-reporter/src/types.ts

```diff
@@ -59,12 +59,19 @@ export type TestCaseSummary = {
   projectName: string;
   location: Location;
   annotations: { type: string, description?: string }[];
-  outcome: 'skipped' | 'expected' | 'unexpected' | 'flaky';
+  outcome: keyof typeof Outcome;
```

Comment:
We are using [string literal types](https://www.typescriptlang.org/docs/handbook/literal-types.html#string-literal-types) instead of enums, which already gives us type checking.

Reply:
You're welcome, but actually its more about the language SourceProp which is not type checked properly. Only for a string.